In [25]:
import pandas as pd

In [26]:
df = pd.read_csv('auto.csv', index_col='ID')
df.head()

CarNumber  Make_n_model  Refund   Fines  History
ID                                                     
0   Y163O8161RUS    Ford Focus     2.0  3200.0      NaN
1    E432XX77RUS  Toyota Camry     1.0  6500.0      NaN
2    7184TT36RUS    Ford Focus     1.0  2100.0      NaN
3   X582HE161RUS    Ford Focus     2.0  2000.0      NaN
4   E34877152RUS    Ford Focus     2.0  6100.0      NaN

In [27]:
df.count()

CarNumber       931
Make_n_model    931
Refund          914
Fines           869
History          82
dtype: int64

In [28]:
df.drop_duplicates(['CarNumber', 'Make_n_model', 'Fines'], keep='last', inplace=True)
df.count()

CarNumber       725
Make_n_model    725
Refund          713
Fines           665
History          65
dtype: int64

In [29]:
df.isna().sum()

CarNumber         0
Make_n_model      0
Refund           12
Fines            60
History         660
dtype: int64

> History column has 660 missing cells, so we drop the column

In [30]:
df.drop('History', axis=1, inplace=True)
df.count()

CarNumber       725
Make_n_model    725
Refund          713
Fines           665
dtype: int64

In [31]:
df

CarNumber    Make_n_model  Refund    Fines
ID                                                
0    Y163O8161RUS      Ford Focus     2.0   3200.0
1     E432XX77RUS    Toyota Camry     1.0   6500.0
2     7184TT36RUS      Ford Focus     1.0   2100.0
3    X582HE161RUS      Ford Focus     2.0   2000.0
5    92918M178RUS      Ford Focus     1.0   5700.0
..            ...             ...     ...      ...
926  Y163O8161RUS      Ford Focus     2.0   1600.0
927  M0309X197RUS      Ford Focus     1.0  22300.0
928  O673E8197RUS      Ford Focus     2.0    600.0
929  8610T8154RUS      Ford Focus     1.0   2000.0
930  H419XE197RUS  Toyota Corolla     2.0      NaN

[725 rows x 4 columns]

In [32]:
df['Refund'].fillna(method='ffill', inplace=True)
df.count()

CarNumber       725
Make_n_model    725
Refund          725
Fines           665
dtype: int64

In [33]:
fines_mean = df.Fines.mean(axis=0, skipna=True)
fines_mean

8594.586466165414

In [34]:
df.Fines.fillna(value=fines_mean, inplace=True)
df.count()

CarNumber       725
Make_n_model    725
Refund          725
Fines           725
dtype: int64

In [43]:
df['Make'] = df['Make_n_model'].apply(lambda x: x.split(' ')[0])
df['Model'] = df['Make_n_model'].apply(lambda x: x.split()[1] if len(x.split()) == 2 else None)
df.drop('Make_n_model', axis=1, inplace=True)
print(df.count())
df.head()

CarNumber    725
Refund       725
Fines        725
Make         725
Model        716
dtype: int64


CarNumber  Refund   Fines    Make  Model
ID                                             
0   Y163O8161RUS     2.0  3200.0    Ford  Focus
1    E432XX77RUS     1.0  6500.0  Toyota  Camry
2    7184TT36RUS     1.0  2100.0    Ford  Focus
3   X582HE161RUS     2.0  2000.0    Ford  Focus
5   92918M178RUS     1.0  5700.0    Ford  Focus

In [46]:
df.to_json('auto.json', orient='records')